src/main/host/thread.c

```diff
@@ -78,13 +78,13 @@ void thread_unref(Thread* thread) {
     }
 }
 
-void thread_run(Thread* thread, gchar** argv, gchar** envv) {
+void thread_run(Thread* thread, gchar** argv, gchar** envv, const char* workingDir) {
```

Comment:
Optional - the thread could get the workingDir from its process instead of wiring it through here. OTOH the same could probably be said of `argv` and `envv`, so maybe it makes sense to pass `workingDir` here for consistency.

Reply:
I'd prefer to keep them consistent. And these members aren't accessible through accessors from the thread, so I think it's better to keep these private to the process.